src/corecel/sys/ScopedProfiling.hh

```diff
@@ -0,0 +1,65 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file corecel/sys/ScopedProfiling.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <string>
+
+#include "celeritas_config.h"
+
+namespace celeritas
+{
+
+//---------------------------------------------------------------------------//
+/*!
+ * Input arguments for the nvtx implementation.
+ */
+struct ScopedProfilingInput
+{
+    // Name of the range
+    std::string name;
+    // ARGB
+    uint32_t color{0xFF00FF00};
+    // User data
+    int32_t payload{0};
+    // Category, used to group ranges together
+    uint32_t category{0};
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * RAII class for scoped profiling.
+ *
+ * Implementations should support multithreaded context where each thread have
+ * one or more alive instance of this class.
+ *
+ * This is useful for wrapping specific code fragment in a range for profiling,
+ * e.g. ignoring of VecGeom instantiation kernels, profiling a specific action
+ * or loop on the CPU.
+ * TODO: Template ScopedProfiling over profiling backend if we need to add a
+ * new one
+ */
+class ScopedProfiling
+{
+  public:
+    using Input = ScopedProfilingInput;
+    // Activate profiling
+    explicit ScopedProfiling(Input input);
+    // Deactivate profiling
+    ~ScopedProfiling();
+    // RAII semantics
+    void* operator new(std::size_t count) = delete;
```

Comment:
I've never seen this pattern before... is it necessary? Do we need to add it to all our RAII classes? For comparison, [scoped_lock](https://en.cppreference.com/w/cpp/thread/scoped_lock) deletes the assignment operator but doesn't delete `new`.

Reply:
It's not technically needed but per the name of the class, dynamically allocating a new instance is not "scoped profiling", so does it make sense to allow it? The C++ API of nvtx does that for their implementation of the scoped_range.

Reply:
I think `delete` and such should be to prevent *accidental* misuses, since it's not possible for us to prevent all *intentional* misuse.

Besides, if wanted to (for example) put it in a `unique_ptr` or `shared_ptr` to change/share the scope at runtime, I think that's a reasonable use case. A dumb user could `new` it and never release it, and then it just becomes an open range? Which seems like the expected behavior.